### PR TITLE
chore: read AZURE_POOL_MANAGEMENT_ENDPOINT from env vars

### PR DESCRIPTION
**Pull request by manekinekko**
# Description

Read `AZURE_POOL_MANAGEMENT_ENDPOINT` from env vars in `AzureCodeInterpreterToolSpec`.

Fixes # (issue): N/A

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py [hunk 1]
```diff
@@ -17,6 +18,8 @@
 
 from llama_index.core.tools.tool_spec.base import BaseToolSpec
 
+dotenv.load_dotenv()
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py:21]: I don't think this should be in the tools class? It should be up to the user to use dotenv (or however they want to load env vars)

Comment (marcusschiesser) [reply llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py:21]: @manekinekko yes, create-llama code will already use `dotenv`

**Review by marcusschiesser — COMMENTED** [review]
@logan-markewich FYI: this PR is to simplify azure tool usage in create-llama

File: llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py (hunk 1 above)
Comment (manekinekko) [reply llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py:21]: should I remove this line?

Comment (marcusschiesser) [reply llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py:21]: yes this one and it's import, then this PR is good to go

Comment (manekinekko) [reply llama-index-integrations/tools/llama-index-tools-azure-code-interpreter/llama_index/tools/azure_code_interpreter/base.py]: done.

**Review by marcusschiesser — APPROVED** [review]

**Comment by marcusschiesser** [comment]
This PR is to add the tool to create-llama (https://github.com/run-llama/create-llama/pull/158)

@seldo there was a typo in the tool's config parameter (pool_managment_endpoint instead of pool_management_endpoint) - this might affect your example code
@logan-markewich if you're ok with the changes, please merge

**Comment by manekinekko** [comment]
@marcusschiesser thanks for catching that typo. Sorry about that 😅

**Pull request merged**